src/core/lv_obj_style.c

```diff
@@ -224,14 +224,14 @@ lv_style_value_t lv_obj_get_style_prop(const lv_obj_t * obj, lv_part_t part, lv_
 {
     lv_style_value_t value_act;
     bool inherit = lv_style_prop_has_flag(prop, LV_STYLE_PROP_INHERIT);
```

Comment:
Do we need to store the  `INHERIT` flag for each property or should it be rather the default value of some properties?

Reply:
We could have a hardcoded default value but I suspect the added code will use up most of the space saved by not having a lookup table for built-in properties.

Reply:
I rather concerned about the inconsistency as this way we can set the inherit flag in 2 places:
- in the [prop look up table](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_style.c#L29) 
- in the property value (with this feature)

Once we have this feature we don't need to use the look up table for this as it causes confusion about where inheritance is/should be set. 

Adding ~10 props [here](https://github.com/lvgl/lvgl/blob/master/src/misc/lv_style.c#L361) shouldn't require too much space but at least there were only one clear way to set inherit flag.

Reply:
These are separate flags. The flag here (`LV_STYLE_PROP_INHERIT`, which will be renamed to `LV_STYLE_PROP_FLAG_INHERIT` in v9) specifies that the property supports inheritance from parents at all (because some like x and y don't). That flag is specified at the property type level.

The other flag (the new one) is effectively a special value for each style that forces `get_prop_core` to immediately check the parent's styles. That is specified in the prop-value pair within a style. It just so happens to be stored with the property ID because there are free bits there.

Reply:
>  The flag here (LV_STYLE_PROP_INHERIT, which will be renamed to LV_STYLE_PROP_FLAG_INHERIT in v9) specifies that the property supports inheritance from parents at all (because some like x and y don't). 

I see. Then what about a name like `LV_STYLE_PROP_FLAG_INHERITABLE`?

Reply:
That name is definitely better, but we can't change it till v9 without breaking user code (unless we keep the old name as well).

Or do you think it's minor enough to change, since it can be fixed with a find-and-replace?

Reply:
We can keep it, I just wanted to be sure we are talking about the same thing. I added a note about the new name here: https://github.com/lvgl/lvgl/pull/3390#discussion_r885915769 

I suggest changing at least the variable name:
```suggestion
    bool inheritable = lv_style_prop_has_flag(prop, LV_STYLE_PROP_INHERIT);
```

Reply:
Changed as you suggested.